scripts/ci/service_name.py

```diff
@@ -0,0 +1,126 @@
+    cmd_table = invoker.commands_loader.load_command_table(None)
+
+    #   turn off applicability check for all loaders
+    for loaders in invoker.commands_loader.cmd_to_loader_map.values():
+        for loader in loaders:
+            loader.skip_applicability = True
+
+    #   filter the command table to only get commands from extensions
+    cmd_table = {k: v for k, v in cmd_table.items() if isinstance(v.command_source, ExtensionCommandSource)}
+    invoker.commands_loader.command_table = cmd_table
+    print('FOUND {} command(s) from the extension.'.format(len(cmd_table)))
+
+    for cmd_name in cmd_table:
+        invoker.commands_loader.load_arguments(cmd_name)
+
+    invoker.parser.load_command_table(invoker.commands_loader)
+
+    # 2. Now load applicable help files
+    parser_keys = []
+    parser_values = []
+    sub_parser_keys = []
+    sub_parser_values = []
+    _store_parsers(invoker.parser, parser_keys, parser_values, sub_parser_keys, sub_parser_values)
+    for cmd, parser in zip(parser_keys, parser_values):
+        if cmd not in sub_parser_keys:
+            sub_parser_keys.append(cmd)
+            sub_parser_values.append(parser)
+    help_ctx = cli_ctx.help_cls(cli_ctx=cli_ctx)
+    help_files = []
+    for cmd, parser in zip(sub_parser_keys, sub_parser_values):
+        try:
+            help_file = GroupHelpFile(help_ctx, cmd, parser) if _is_group(parser) \
+                else CliCommandHelpFile(help_ctx, cmd, parser)
+            help_file.load(parser)
+            help_files.append(help_file)
+        except Exception as ex:
+            print("Skipped '{}' due to '{}'".format(cmd, ex))
+    help_files = sorted(help_files, key=lambda x: x.command)
+    return help_files
+
+
+def _store_parsers(parser, parser_keys, parser_values, sub_parser_keys, sub_parser_values):
+    for s in parser.subparsers.values():
+        parser_keys.append(_get_parser_name(s))
+        parser_values.append(s)
+        if _is_group(s):
+            for c in s.choices.values():
+                sub_parser_keys.append(_get_parser_name(c))
+                sub_parser_values.append(c)
+                _store_parsers(c, parser_keys, parser_values, sub_parser_keys, sub_parser_values)
+
+
+def _get_parser_name(s):
+    return (s._prog_prefix if hasattr(s, '_prog_prefix') else s.prog)[3:]
+
+
+def _is_group(parser):
+    return getattr(parser, '_subparsers', None) is not None \
+        or getattr(parser, 'choices', None) is not None
+
+
+def check():
+    az_cli = AzCli(cli_name='az',
+                   commands_loader_cls=MainCommandsLoader,
+                   invocation_cls=AzCliCommandInvoker,
+                   parser_cls=AzCliCommandParser,
+                   help_cls=AzCliHelp)
+    help_files = get_extension_help_files(az_cli)
+    high_command_set = set()
+    for help_file in help_files:
+        if help_file.command:
+            high_command_set.add(help_file.command.split()[0])
+    print('high_command_set:')
```

Comment:
Is `high command` an existing term or coined by you? Is it OK to use `command group` to refer to the same thing?

Reply:
The left most word in a command, e.g., `vm`, `disk`. Let me add a comment.